### PR TITLE
Add health check doc

### PR DESCRIPTION
**Pull request by lixiaoyuner**
Add container upgrade via k8s health check probe design doc.

### CODE REVIEW

File: doc/kubernetes/health-check.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Why need health-check
+- Current status
+    - We can only check whether the k8s scheduled container is running or not, couldn't exactly know the real supervisor service inside container is healthy or not
+- Problems should be resolved
+    - Health-check should check whether the real supervisor service is healthy or not
+    - If not healthy, there better be different error codes to help debug
+
+# How do we Implement the health-check 
+- Leverage k8s probe tool
+    - K8s has three kinds of probe, [liveness, readiness and startup probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes) which can help us probe state inside container. we can use startup probe for our scenario.
```
Comment (losha228) [inline doc/kubernetes/health-check.md:10]: There are 3 kinds of probes to handle container failure during different stages.  Could you please describe how sonic k8s handle the 3 situations now for sonic container ?

File: doc/kubernetes/health-check.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    - Startup probe has four probe types, http|TCP|gRPC|command, command type should be good for our scenario.
```
Comment (losha228) [inline doc/kubernetes/health-check.md:11]: not only startup probe, each probe supports those probe types.

File: doc/kubernetes/health-check.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    - Command type means that we could deploy one script inside container, k8s will call this script and get the exit code when start the container. If the script exit code is zero, k8s thinks it's healthy. If the script exit code is non-zero, k8s think it's not healthy. We can use different non-zero exit codes to recognize what the issue is if probe failed.
+    - Start command probe example:
+    - ![](startup_probe.jpg)
+
+# How do we implement the health-check script inside container
+- Script path and name
+    - /usr/bin/readiness_probe.sh
+- Health-Check logic in the script(two steps for now)
+    - Do common checks which are same for all containers
+        - Currently, one common check is that if the supervisor start.sh exists, we need to ensure that it exits with code 0.
```
Comment (losha228) [inline doc/kubernetes/health-check.md:21]: it's better to add a description for start.sh, each sonic container has a start.sh which is for service initialization.

File: doc/kubernetes/health-check.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        - Will not check critical services, because these services should be managed by the exit-listener supervisor service.
```
Comment (losha228) [inline doc/kubernetes/health-check.md:22]: this is how sonic handle liveness probe.

File: doc/kubernetes/health-check.md (hunk 1 above)
Comment (lixiaoyuner) [reply doc/kubernetes/health-check.md:10]: I think the startup probe is enough for our scenario.

File: doc/kubernetes/health-check.md (hunk 2 above)
Comment (lixiaoyuner) [reply doc/kubernetes/health-check.md:11]: Yes, I mean we plan use startup probe, so introduce the startup probe only. More details can be found in the link.

**Review by losha228 — APPROVED** [review]

**Pull request merged**